src/hera/events/service.py

```diff
@@ -73,8 +74,21 @@ def format_token(t):
         else:
             self.token = None
 
+        use_session = use_session if use_session is not None else global_config.use_session
+        if use_session:
+            self.session: Optional[requests.Session] = requests.Session()
+        else:
+            self.session = None
+
         self.namespace = namespace or global_config.namespace
 
+    def request(self, method, **kwargs):
```

Comment:
I'd mark this private; I don't think we want this added to the public API. (Same for WorkflowsService.)
```suggestion
    def _request(self, method, **kwargs):
```